lib/core/proxy.js

```diff
@@ -8,46 +9,50 @@ exports.serve = function(ipc, host, port, ws){
 
     let proxy = httpProxy.createProxyServer({
         target: {
-            host: host,
-            port: port + 10
+            host,
+            port: port + constants.blockchain.servicePortOnProxy
           },
         ws: ws
     });
 
     proxy.on('error', function () {
-        proxy.close();
+        console.log(__("Error forwarding requests to blockchain/simulator"));
+        process.exit();
     });
 
     proxy.on('proxyRes', (proxyRes) => {
         let resBody = [];
         proxyRes.on('data', (b) => resBody.push(b));
         proxyRes.on('end', function () {
             resBody = Buffer.concat(resBody).toString();
+
+            let jsonO;
             try {
-                let jsonO = JSON.parse(resBody);
-                if(commList[jsonO.id]){
-                    commList[jsonO.id].transactionHash = jsonO.result;
-                    transactions[jsonO.result] = {commListId: jsonO.id};
-                } else if(receipts[jsonO.id]){
-                    commList[receipts[jsonO.id]].blockNumber = jsonO.result.blockNumber;
-                    commList[receipts[jsonO.id]].gasUsed = jsonO.result.gasUsed;
-                    commList[receipts[jsonO.id]].status = jsonO.result.status;
-                   
-                    if(ipc.connected && !ipc.connecting){
-                        ipc.request('log', commList[receipts[jsonO.id]]);
-                    } else {
-                        ipc.connecting = true;
-                        ipc.connect(() => {
-                            ipc.connecting = false;
-                        });
-                    }
+                jsonO = JSON.parse(resBody);
+            } catch(e) {
+                return;
```

Comment:
You should print the error so we know something went wrong

Reply:
In this case, not all the responses received are JSON Objects. I'm using JSON.parse as a way to verify if it's a json object, and then continue the function execution. Is there a better approach to identify if a string is a valid json object?

Reply:
Maybe there is, but I can't think of one right now, and I thin it's ok then